Написать API класса Треугольник#1

anythinggwp opened the pull request:
1. API Triangle
2. Написана документация к методам класса

anythinggwp committed:
Написать API класса Треугольник#1

1. API Triangle
2.Написана документация к методам класса

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+#pragma once
```
Переименуйте файл из `Header` &rarr; `Triangle`

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    double b;
+    double c;
+    //��������
+    double p;
```
Напишите явно `perimetr`, `square`. Это не поля, а меоды
Высоты - не поля, а методы. Названия не нравятся, без комментариев не ясно

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    Triangle();
+    //������ ������� �
+    //���������� �
+    double Geta()
```
реализацию здесь не пишут. Только `API`
Про реализацию... сторона может быть `<= 0`?

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    double hc;
+public:
+    //����������� �� ���������
+    Triangle();
```
я бы запретила создавать треугольник со сторонами (0, 0, 0)

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    }
+    //������ ������� �
+    //��������� �������� ������� a
+    void Seta(float valuea)
```
не нравится найминг, определитесь со стилем [snake_case](https://ru.wikipedia.org/wiki/Snake_case) или [CamelCase](https://en.wikipedia.org/wiki/Camel_case)

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    //���������� ��������
+    double Getp()
+    {
+        p = a + b + c;
```
`return a + b + c;`

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    //���������� �������
+    double Gets()
+    {
+        float s1 = Getp() / 2;
```
почему `float`?
почему `s1`? как это связано с "полупериметром"?

MarinaPaley commented on objective1/objective1/objective1/Header.h:
```diff
@@ -0,0 +1,112 @@
+    {
+        if (a == c & a == b)
+        {
+            cout << "��� ������������ ��������������" << endl;
```
Вы ВНИМАТЕЛЬНО слушали лекцию(!
`API` никак не связано с реализацией. Реализация, если это ТОЛЬКО `template` будет находится в `header` класса, но не в вашем случае!
связь с консолью, файлами в библиотеке не должна быть. Библиотека никак не связана с потоком ввода-вывода!!!

anythinggwp committed:
Изменения после проверки работы (API Triangle)

1. Изменено название Хедера
2. Добавлен дополнительный конструктор класса
3.Изменён нейминг функций в классе
4.Убраны лишние переменный и параметры объекта
5.Убраны взаимодействия класса с консолью

anythinggwp committed:
Изменения после проверки работы (API Triangle)

1. Изменено название Хедера
2. Добавлен дополнительный конструктор класса
3.Изменён нейминг функций в классе
4.Убраны лишние переменный и параметры объекта
5.Убраны взаимодействия класса с консолью

MarinaPaley reviewed (changes requested):
Вы не исправили _все_ замечания

anythinggwp committed:
Написать API класса Треугольник#1

1. API Triangle
2.Написана документация к методам класса

anythinggwp committed:
Написать API класса Треугольник#1

1. API Triangle
2.Написана документация к методам класса

MarinaPaley commented on objective1/objective1/objective1/Triangle.cpp:
```diff
@@ -1,41 +1,42 @@
+string Triangle::mod_triangle()
 {
-    if (_a==_c & _a==_b)
+    if ((double_compare(_a,_b,0.000000012)== 1) && (double_compare(_a,_c, 0.000000012)==1))
```
`numeric_limits<double>::min();`

MarinaPaley commented on objective1/objective1/objective1/Triangle.cpp:
```diff
@@ -1,41 +1,42 @@
-    if (_a!=_b & _b!=_c &_a !=_c)
+    if (double_compare(_a, _b, 0.000000012)==0 && double_compare(_a, _c, 0.000000012) == 0 && double_compare(_b, _c, 0.000000012) == 0)
     {
         return "�������������";
```
`UTF-8` кодировка!

MarinaPaley commented on objective1/objective1/objective1/helper.cpp:
```diff
@@ -0,0 +1,34 @@
+}
+bool Triangle::proportion(const double value_a,const double value_b,const double value_c)
+{
+	if (value_a<(value_b+value_c))
```
`return a < (b + c) && b < (a + c) && c < (a + b);`

MarinaPaley commented on objective1/objective1/objective1/helper.cpp:
```diff
@@ -0,0 +1,34 @@
+#include "Triangle.h"
+bool Triangle::double_compare(const double value_a,const double value_b,const double eps) 
+{
+	if (abs(value_a - value_b) < eps) 
```
`return abs(value_a - value_b) <= numeric_limits<double::min();`

MarinaPaley commented on objective1/objective1/objective1/Triangle.cpp:
```diff
@@ -1,41 +1,42 @@
+Triangle::Triangle(const double value_a,const double value_b, const double value_c)
 {
-    if (valuea != 0 && valueb != 0 && valuec != 0 && valuea > 0 && valueb > 0 && valuec > 0)
+    if(proportion(value_a,value_b,value_c)==1)
```
`if (!proportion(value_a,value_b,value_c)
throw ("Треугольник не существует");`

anythinggwp committed:
Написать API класса Треугольник#1

1. API Triangle
2.Написана документация к методам класса
